File: GamesDB/GamesDB/DBLayer/Game.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GamesDB
{
    class Game
    {
        private String _nome;
        private String _jogoid;
        private String _categoria;
        private String _rating;
        private String _datalancamento;
        private String _especificacoes;

        public string nome
        {
            get { return _nome; }
            set { _nome = value; }
        }

        public string jogoid
        {
            get { return _jogoid; }
            set { _jogoid = value; }
        }

        public string categoria
        {
            get { return _categoria; }
            set { _categoria = value; }
        }
        public string rating
        {
            get { return _rating; }
            set { _rating = value; }
        }

        public string datalancamento
        {
            get { return _datalancamento; }
            set { _datalancamento = value; }
        }

        public string especificacoes
        {
            get { return _especificacoes; }
            set { _especificacoes = value; }
        }

        public override string ToString()
        {
            return String.Format("{0,-30}\t{1,-30}\t{2,-37}\t{3,-20}", _nome, _jogoid, _categoria, _rating, _datalancamento, _especificacoes);
        }

    }
}

File: GamesBD/GamesBD/Game.vb
<Serializable()> Public Class Game
    Private _Nome As String
    Private _JogoID As String
    Private _Preco As String
    Private _DataLancamento As String
    Private _Rating As String
    Private _Categoria As String
    Private _EspecifacoesTecnicas As String

    Property Nome() As String
        Get
            Return _Nome
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Nome não pode estar vazio")
                Exit Property
            End If
            _Nome = value
        End Set
    End Property

    Property JogoID() As String
        Get
            Return _JogoID
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo JogoID não pode estar vazio")
                Exit Property
            End If
            _JogoID = value
        End Set
    End Property

    Property Preco() As String
        Get
            Return _Preco
        End Get
        Set(value As String)
            _Preco = value
        End Set
    End Property

    Property DataLancamento() As String
        Get
            Return _DataLancamento
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Data de Lançamento não pode estar vazio")
                Exit Property
            End If
            _DataLancamento = value
        End Set
    End Property

    Property Rating() As String
        Get
            Return _Rating
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Rating não pode estar vazio")
                Exit Property
            End If
            _Rating = value
        End Set
    End Property

    Property Categoria() As String
        Get
            Return _Categoria
        End Get
        Set(value As String)
            If value Is Nothing Or value = "" Then
                Throw New Exception("Campo Categoria não pode estar vazio")
                Exit Property
            End If
            _Categoria = value

        End Set
    End Property

    Property EspecificacoesTecnicas() As String
        Get
            Return _EspecifacoesTecnicas
        End Get
        Set(value As String)
            _EspecifacoesTecnicas = value
        End Set
    End Property

    Overrides Function ToString() As String
        Return _JogoID & "   " & _Rating & "     " & _Nome
    End Function

    Public Sub New()
        MyBase.New()
    End Sub

    Public Sub New(ByVal Nome As String)
        MyBase.New()
        Me.Nome = Nome
    End Sub

End Class
